scripts/start_update.sh

```diff
@@ -5,8 +5,199 @@
 # Copyright (C) 2018 Raymond Siudak <raysiudak@gmail.com>
 #
 # SPDX-License-Identifier: AGPL-3.0+
-#
-# This script is a placeholder that is executed as entry point for system configuration routines.
 
-echo 'Executing test update script'
+set -eu
+
+# Script mandatory arguments
+# The mount point of the update toolbox, used as chroot root.
+TOOLBOX_MOUNT=""
+# The mount point where the update archive and this toolbox where found
+UPDATE_MOUNT=""
+# The target update storage device
+TARGET_STORAGE_DEVICE=""
+
+# Toolbox execution environment arguments
+# The directory in the chroot environment containing the system update and configuration files.
+SYSTEM_UPDATE_DIR="${SYSTEM_UPDATE_DIR:-/etc/system_update}"
+# The partition table file to use, default jedi_emmc_sfdisk.table, should exist in the system update dir.
+PARTITION_TABLE_FILE="${PARTITION_TABLE_FILE:-jedi_emmc_sfdisk.table}"
+# The exclude file list when updating the firmware files, should exist in the system update dir.
+UPDATE_EXCLUDE_LIST_FILE="${UPDATE_EXCLUDE_LIST_FILE:-jedi_update_exclude_list.txt}"
+# The directory in the chroot environment containing the source update files.
+UPDATE_ROOTFS_SOURCE="/mnt/update_rootfs_source"
+
+update_rootfs_archive=""
+
+
+usage()
+{
+    echo "Usage: ${0} [OPTIONS] <TOOLBOX_MOUNT> <UPDATE_MOUNT> <TARGET_STORAGE_DEVICE>"
+    echo "This is the update entry point script, it is responsible for setting up the"
+    echo "environment in which the update toolbox can be used to configure and update"
+    echo "the firmware."
+    echo "  -h Print this help text and exit"
+}
+
+cleanup()
+{
+    if [ -d "${TOOLBOX_MOUNT}/${UPDATE_ROOTFS_SOURCE}" ]; then
+        echo "Cleaning up, removing '${TOOLBOX_MOUNT}/${UPDATE_ROOTFS_SOURCE}/' files."
+        rm -rf "${TOOLBOX_MOUNT:?}/${UPDATE_ROOTFS_SOURCE:?}"/*
+    fi
+
+    if [ -f "${UPDATE_MOUNT}/${update_rootfs_archive}" ]; then
+        echo "Cleaning up, removing '${UPDATE_MOUNT}/${update_rootfs_archive}'."
+        rm "${UPDATE_MOUNT:?}/${update_rootfs_archive:?}"
+    fi
+
+    if grep -q "${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}" /proc/mounts; then
+        echo "Cleaning up, unmount: '${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}'."
+        umount "${TOOLBOX_MOUNT}${UPDATE_ROOTFS_SOURCE}"
+    fi
+}
+
+extract_update_rootfs()
```

Comment:
I kinda would have expected this to happen as 40_extract_rootfs.sh or as part of the 50_update_files.sh instead of outside the apply update scripts?